app/src/main/java/com/rahobbs/todo/fragments/TodoFragment.java

```diff
@@ -0,0 +1,228 @@
+    private TextView mDueDateTextField;
+    private CheckBox mCompletedCheckbox;
+    private LinearLayout mDateComponents;
+
+    public static TodoFragment newInstance(UUID todoId) {
+        Bundle args = new Bundle();
+        args.putSerializable(ARG_TODO_ID, todoId);
+        TodoFragment fragment = new TodoFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Firebase.setAndroidContext(getContext());
+        UUID todoId = (UUID) getArguments().getSerializable(ARG_TODO_ID);
+        setHasOptionsMenu(true);
+
+        if (todoId == null) {
+            throw new RuntimeException("Todo ID is null.");
+        }
+
+        mTodo = TodoLab.get(getActivity()).getItem(todoId);
+        Log.v("DETAIL_CREATED", mTodo.getID().toString());
+
+        if (mTodo == null) {
+            throw new RuntimeException("No Todo item with provided ID (" + todoId.toString() + ") was found.");
+        }
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.fragment_todo_item, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_send_feedback:
+                Intent i = new Intent(Intent.ACTION_SEND);
+                Feedback fb = new Feedback();
+                fb.sendFeedback(i);
+                startActivity(Intent.createChooser(i, "Send mail..."));
+                return true;
+            case R.id.menu_item_delete_todo:
+                UUID todoId = mTodo.getID();
+                TodoLab.get(getActivity()).deleteTodoItem(todoId);
+
+                String titleToDisplay;
+                if (mTodo.getTitle() == null) {
+                    titleToDisplay = "Unnamed task";
+                } else if (mTodo.getTitle().length() > 25) {
+                    titleToDisplay = "\"" + mTodo.getTitle().trim().substring(0, 25) + "\"";
+                } else {
+                    titleToDisplay = "\"" + mTodo.getTitle().trim() + "\"";
+                }
+
+                Toast.makeText(getActivity(),
+                        titleToDisplay + " deleted",
+                        Toast.LENGTH_SHORT).show();
+                getActivity().finish();
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @Override
+    public void onPause() {
+        saveNewItem();
+        Log.v("DETAIL_PAUSE", mTodo.getID().toString());
+        super.onPause();
+    }
+
+    private void saveNewItem() {
+        if (mTodo.getTitle() == null) {
+            UUID mTodoID = mTodo.getID();
+            TodoLab.get(getActivity()).deleteTodoItem(mTodoID);
+        }
+
+        TodoLab.get(getActivity()).updateItem(mTodo);
```

Comment:
how come we are updating not adding? I haven't looked at the `update` method yet.. so it might handle exisiting items. in which case ignore this comment